django/db/models/sql/compiler.py

```diff
@@ -281,6 +281,7 @@ def get_order_by(self):
                 if not isinstance(field, OrderBy):
                     field = field.asc()
                 if not self.query.standard_ordering:
+                    field = field.copy()
```

Comment:
Only question would be whether `reverse_ordering()` should do the copying internally or not, rather than the copying happening outside. Long time since I've worked on this code, and I'm guessing your day spent debugging this could answer better than I :)

Reply:
Current behavior of `reverse_ordering()` is [documented](https://docs.djangoproject.com/en/stable/ref/models/expressions/#django.db.models.Expression.reverse_ordering) and I can imagine that someone is using this without cloning so I would like to leave it as it is.